### PR TITLE
ci: update workflows and sync-files

### PR DESCRIPTION
**Pull request by xmfcx**
## Description

Merged the sync-files PR generated by this PR's branch:
- https://github.com/autowarefoundation/autoware.universe/pull/7368

This PR removes the:
- https://github.com/autowarefoundation/autoware.universe/blob/main/.github/workflows/spell-check-partial.yaml
- Now it's managed by the sync-files action as 2 new files (already merged):
  - https://github.com/autowarefoundation/autoware.universe/blob/main/.github/workflows/spell-check-daily.yaml
  - https://github.com/autowarefoundation/autoware.universe/blob/main/.github/workflows/spell-check-differential.yaml

- All `build-and-test` derivatives are removed from the sync-files list and now being managed manually. (all the sd commands became a bit too ridiculous)

Current state of `build-and-test` derivatives:

### `build-and-test-daily.yaml`

- Runs daily or on `workflow_dispatch`.
- Runs for `cuda` and `non-cuda`.
- `fetch-depth: 1`
- Uploads coverage with flag `total`

### `build-and-test-daily-arm.yaml`

- Runs daily or on `workflow_dispatch`.
- Runs for `cuda` and `non-cuda`.
- `fetch-depth: 1`
- Uploads coverage with flag `total-arm`

### `build-and-test.yaml`

- Runs per commit into main.
- Runs for `cuda` only.
- `fetch-depth: 1`
- Uploads coverage with flag `total`

### `build-and-test-differential.yaml`

- Runs per PR.
- Runs for `cuda` and `non-cuda`.
- Fetch only PR commits.
- Uploads coverage with flag `differential`
- Now uses new `make-sure-label-is-present` workflow, supporting multiple labels
   - requires label: `tag:run-build-and-test-differential`
- Runs on `reopened` event too.

### `build-and-test-differential-arm64.yaml`

- Runs per PR.
- Runs for `cuda` and `non-cuda`.
- Fetch only PR commits.
- Uploads coverage with flag `differential-arm64`
- Now uses new `make-sure-label-is-present` workflow, supporting multiple labels
   - requires label: `type:arm64`
- Runs on `reopened` event too.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: .github/workflows/build-and-test-cuda.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  build-and-test-cuda:
+    runs-on: [self-hosted, linux, X64]
+    container: ${{ matrix.container }}${{ matrix.container-suffix }}
+    strategy:
+      fail-fast: false
+      matrix:
+        rosdistro:
+          - humble
+        container-suffix:
+          - -cuda
```
Comment (mitsudome-r) [inline .github/workflows/build-and-test-cuda.yaml:20]: NOTE: 
We assume that we should be able to catch most of build failures in build-and-test-differential so we will only run against cuda here.

**Comment by xmfcx** [comment]
image

Ok now it fails on failure. I will test again to see if it passes on success.

File: .github/workflows/build-and-test.yaml [hunk 2]
```diff
@@ -1,14 +1,13 @@
-name: build-and-test
+name: build-and-test-daily
```
Comment (xmfcx) [inline .github/workflows/build-and-test.yaml:1]:
```suggestion
name: build-and-test
```

File: .github/workflows/build-and-test.yaml [hunk 3]
```diff
@@ -1,14 +1,13 @@
...
-  schedule:
-    - cron: 0 0 * * *
+    branches:
+      - main
   workflow_dispatch:
 
 jobs:
-  build-and-test:
-    if: ${{ github.event_name != 'push' || github.ref_name == github.event.repository.default_branch }}
+  build-and-test-daily:
```
Comment (xmfcx) [inline .github/workflows/build-and-test.yaml:10]:
```suggestion
  build-and-test:
```

**Review by mitsudome-r — APPROVED** [review]

**Pull request merged**